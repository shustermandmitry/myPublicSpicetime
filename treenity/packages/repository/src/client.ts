export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
