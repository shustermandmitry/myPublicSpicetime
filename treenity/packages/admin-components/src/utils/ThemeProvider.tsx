/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
