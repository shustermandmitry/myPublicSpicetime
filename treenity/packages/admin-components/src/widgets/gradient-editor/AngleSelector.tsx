import Button from '@/components/button';
import Icon from '@/components/icon';
import Input from '@/components/input';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { useToggle } from '@treenity/ui-kit/hooks';
import { omitProps } from '@treenity/ui-kit/utils';
import React, { ChangeEvent, FC, useRef } from 'react';

interface IAngleSelector {
  onChange(value: number): void;
  value: number;
}

const DISTRICT_DEGREES = 360;

const AngleSelector: FC<IAngleSelector> = ({ onChange, value }) => {
  const dialRef = useRef<HTMLDivElement>(null);
  const [toggle, setToggle] = useToggle();

  const onInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = Number(e.target.value);
    onChange(newValue);
  };

  const rotateIncrease = () => {
    onChange((value + 1) % DISTRICT_DEGREES);
  };

  const rotateDecrease = () => {
    onChange((value - 1 + DISTRICT_DEGREES) % DISTRICT_DEGREES);
  };

  const getAngleFromMousePosition = (event: MouseEvent) => {
    if (!dialRef.current) return 0;
    const rect = dialRef.current.getBoundingClientRect();
    const centerX = rect.left + rect.width / 2;
    const centerY = rect.top + rect.height / 2;
    const deltaX = event.clientX - centerX;
    const deltaY = event.clientY - centerY;
    const rad = Math.atan2(deltaY, deltaX);
    const deg = rad * (180 / Math.PI);
    return Math.floor((deg + 90 + DISTRICT_DEGREES) % DISTRICT_DEGREES);
  };

  const onMouseDown = (event: React.MouseEvent) => {
    setToggle(true);
    onChange(getAngleFromMousePosition(event.nativeEvent));
  };

  const onMouseMove = (event: MouseEvent) => {
    if (toggle) {
      onChange(getAngleFromMousePosition(event));
    }
  };

  const onMouseUp = () => {
    setToggle(false);
  };

  return (
    <Root>
      <OuterCircle
        data-testid="asdqwe"
        angle={value}
        ref={dialRef}
        onMouseDown={onMouseDown}
        onMouseMove={(e: React.MouseEvent<HTMLDivElement>) => onMouseMove(e.nativeEvent)}
        onMouseLeave={onMouseUp}
        onMouseUp={onMouseUp}
      >
        <InnerCircle>
          <Arrow>
            <Icon name="preview_filled" color="primary" />
            <Icon name="color-picker_filled" color="primary" />
            <Circle />
          </Arrow>
        </InnerCircle>
      </OuterCircle>
      <Input
        type="number"
        size="x-small"
        placeholder="0"
        min={0}
        max={DISTRICT_DEGREES}
        value={value}
        onChange={onInputChange}
      />
      <Buttons>
        <Button
          type="secondary-outlined"
          size="x-small"
          icon={<Icon name="arrow-rotate-left_outlined" />}
          onClick={rotateDecrease}
        />
        <Button
          type="secondary-outlined"
          size="x-small"
          icon={<Icon name="arrow-rotate-right_outlined" />}
          onClick={rotateIncrease}
        />
      </Buttons>
    </Root>
  );
};

const Circle = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 4px;
  height: 4px;
  border-radius: 4px;
  background-color: ${p => p.theme.colorPrimary};
`;

const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 4px;
  width: 64px;
`;

const InnerCircle = styled.div`
  position: relative;
  width: 46px;
  height: 46px;
  border: 1px solid ${p => p.theme.gray400};
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const OuterCircle = styled('div', omitProps('angle'))<{ angle: number }>`
  position: relative;
  width: 64px;
  height: 64px;
  border: 1px solid ${p => p.theme.gray400};
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  background: ${p => p.theme.colorBgBase};

  ${p => css`
    transform: rotate(${p.angle - 90}deg);
  `}
`;

const Arrow = styled.div`
  width: 46px;
  height: 1px;
  background-color: ${p => p.theme.colorPrimary};
  position: absolute;

  span {
    position: absolute;
    top: 50%;

    &:nth-of-type(1) {
      transform: translate(90%, -50%);
      right: 1px;
    }

    &:nth-of-type(2) {
      left: 0;
      transform: translate(-39%, -47%) rotate(45deg);
    }

    i {
      font-size: 6px !important;
    }
  }
`;

const Buttons = styled.div`
  display: flex;
  gap: 2px;
  width: 100%;

  && > button {
    width: 100% !important;

    i {
      color: ${p => p.theme.colorText};
    }
  }
`;

export default AngleSelector;
