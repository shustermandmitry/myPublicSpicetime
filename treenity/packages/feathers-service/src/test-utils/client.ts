import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
