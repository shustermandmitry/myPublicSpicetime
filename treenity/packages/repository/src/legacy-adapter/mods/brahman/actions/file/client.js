import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
