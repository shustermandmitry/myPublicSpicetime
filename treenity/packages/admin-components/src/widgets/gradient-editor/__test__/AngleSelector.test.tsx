/*
 * Copyright (c) 2024. Treenity Inc.
 */
import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { act, fireEvent, render } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
import AngleSelector from '../AngleSelector';

const defaultValue: number = 0;

const cookies = createCookies(parseCookies(document.cookie), setCookie);
let valueMock: number = defaultValue;

const updateMock = jest.fn((value: number) => {
  valueMock = value;
  return value;
});

describe('AngleSelector', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = defaultValue;
  });

  it('render AngleSelector component', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <AngleSelector onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-6qlw1o"]`);
    const outerCircle = root!.querySelector(`div[class="css-rp54ss"]`);
    const input = root!.querySelector(`input[class^="ant-input"]`);
    const buttonContainer = root!.querySelector(`div[class="css-5t8zso"]`);
    const buttons = buttonContainer!.querySelectorAll(`button[class^="ant-btn"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(outerCircle).toBeInTheDocument();
    expect(outerCircle).toHaveStyle(`transform: rotate(-90deg)`);
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue(0);
    expect(buttonContainer).toBeInTheDocument();

    [...buttons].forEach(button => {
      const buttonIcon = root!.querySelector(`i`);
      expect(button).toBeInTheDocument();
      expect(buttonIcon).toBeInTheDocument();
    });
  });

  it('change arrow', async () => {
    const { container, rerender } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <AngleSelector onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-6qlw1o"]`);
    const outerCircle = root!.querySelector(`div[class="css-rp54ss"]`);
    const input = root!.querySelector(`input[class^="ant-input"]`);
    const buttonContainer = root!.querySelector(`div[class="css-5t8zso"]`);
    const buttons = buttonContainer!.querySelectorAll(`button[class^="ant-btn"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(outerCircle).toBeInTheDocument();
    expect(outerCircle).toHaveStyle(`transform: rotate(-90deg)`);
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue(0);
    expect(buttonContainer).toBeInTheDocument();
    expect(buttons).toHaveLength(2);

    [...buttons].forEach((button, index) => {
      const buttonIcon = root!.querySelector(`i`);
      expect(button).toBeInTheDocument();
      expect(buttonIcon).toBeInTheDocument();
    });

    act(() => {
      fireEvent.mouseDown(outerCircle!, { clientX: 10, clientY: 10 });
      fireEvent.mouseMove(outerCircle!, { clientX: 20, clientY: 40 });
      fireEvent.mouseUp(outerCircle!, { clientX: 20, clientY: 40 });
    });

    rerender(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <AngleSelector onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(input).toHaveValue(153);
    expect(updateMock.mock.calls).toHaveLength(2);
    expect(valueMock).toBe(153);
    expect(updateMock.mock.calls.at(-1)).toEqual([153]);
    expect(outerCircle).toHaveStyle(`transform: rotate(63deg)`);

    fireEvent.change(input!, { target: { value: '100' } });

    rerender(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <AngleSelector onChange={updateMock} value={valueMock} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(updateMock.mock.calls.at(-1)).toEqual([100]);
    expect(outerCircle).toHaveStyle(`transform: rotate(10deg)`);

    fireEvent.click(buttons[0]!);
    expect(updateMock.mock.calls.at(-1)).toEqual([99]);
    expect(outerCircle).toHaveStyle(`transform: rotate(10deg)`);

    fireEvent.click(buttons[1]!);
    fireEvent.click(buttons[1]!);
    expect(updateMock.mock.calls.at(-1)).toEqual([101]);
    expect(outerCircle).toHaveStyle(`transform: rotate(10deg)`);
  });
});
