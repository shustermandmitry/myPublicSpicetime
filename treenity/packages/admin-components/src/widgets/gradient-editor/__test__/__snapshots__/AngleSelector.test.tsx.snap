// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AngleSelector change arrow 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  width: 64px;
}

.emotion-1 {
  position: relative;
  width: 64px;
  height: 64px;
  border: 1px solid #E9E8EE;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-2 {
  position: relative;
  width: 46px;
  height: 46px;
  border: 1px solid #E9E8EE;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  width: 46px;
  height: 2px;
  background-color: #27ae60;
  position: absolute;
}

.emotion-3 span {
  position: absolute;
  top: 50%;
}

.emotion-3 span:nth-of-type(1) {
  -webkit-transform: translate(90%, -50%);
  -moz-transform: translate(90%, -50%);
  -ms-transform: translate(90%, -50%);
  transform: translate(90%, -50%);
  right: 1px;
}

.emotion-3 span:nth-of-type(2) {
  left: 0;
  -webkit-transform: translate(-90%, -50%) rotate(45deg);
  -moz-transform: translate(-90%, -50%) rotate(45deg);
  -ms-transform: translate(-90%, -50%) rotate(45deg);
  transform: translate(-90%, -50%) rotate(45deg);
}

.emotion-3 span i {
  font-size: 6px!important;
}

.emotion-4 [class^='icon-'],
.emotion-4 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-5 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #27ae60;
}

.emotion-9 {
  height: 20px;
  font-size: 10px;
  border-radius: 4px;
  padding-inline: 6px;
}

.emotion-9.emotion-9 .ant-input-prefix {
  -webkit-margin-end: 6px;
  margin-inline-end: 6px;
}

.emotion-9.emotion-9 i {
  font-size: 12px!important;
}

.emotion-9[type='number']::-webkit-inner-spin-button,
.emotion-9 input[type='number']::-webkit-inner-spin-button {
  display: none;
}

.emotion-9.ant-input {
  font-weight: 500!important;
}

.emotion-9 .ant-input-prefix {
  -webkit-margin-end: 8px;
  margin-inline-end: 8px;
}

.emotion-9 i {
  color: #d9d9d9;
  -webkit-transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-9.ant-input-affix-wrapper-focused i {
  color: #27ae60;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) {
  background-color: #eb57570a;
  border-width: 1px;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) i {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i:before,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) i:before {
  content: '\\e91c';
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input {
  background: transparent;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-moz-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::-moz-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input,
.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::placeholder,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::placeholder {
  color: #eb5757;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
  width: 100%;
}

.emotion-10.emotion-10>button {
  width: 100%!important;
}

.emotion-10.emotion-10>button i {
  color: rgba(38, 35, 63, 0.88);
}

.emotion-12 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  background: #FFF;
  color: #26233F;
  border-color: #E9E8EE;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-12>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-12.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-12 i {
  color: #27ae60;
}

.emotion-12.ant-btn-default:not(:disabled):not(.emotion-12.ant-btn-disabled):hover {
  background: #FFF;
  color: #47BA73;
  border-color: #E5E4EC;
}

.emotion-12.ant-btn-default:not(:disabled):not(.emotion-12.ant-btn-disabled):active {
  background: #FFF;
  color: #178749;
  border-color: #E1DFE8;
}

.emotion-12 i {
  font-size: 16px;
}

.emotion-12>span {
  gap: 4px;
}

.emotion-12,
.emotion-12 span {
  line-height: 10px;
}

.emotion-12.emotion-12>span>span>i {
  font-size: 12px;
}

.emotion-14 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
    data-testid="asdqwe"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-preview_filled emotion-5"
          />
        </span>
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-color-picker_filled emotion-5"
          />
        </span>
      </div>
    </div>
  </div>
  <input
    class="ant-input ant-input-sm emotion-8 ant-input-outlined emotion-9"
    max="360"
    min="0"
    placeholder="0"
    type="number"
    value="0"
  />
  <div
    class="emotion-10"
  >
    <button
      class="ant-btn emotion-8 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-12"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-arrow-rotate-left_outlined emotion-14"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-8 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-12"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-arrow-rotate-right_outlined emotion-14"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`AngleSelector render AngleSelector component 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  width: 64px;
}

.emotion-1 {
  position: relative;
  width: 64px;
  height: 64px;
  border: 1px solid #E9E8EE;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-2 {
  position: relative;
  width: 46px;
  height: 46px;
  border: 1px solid #E9E8EE;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  width: 46px;
  height: 2px;
  background-color: #27ae60;
  position: absolute;
}

.emotion-3 span {
  position: absolute;
  top: 50%;
}

.emotion-3 span:nth-of-type(1) {
  -webkit-transform: translate(90%, -50%);
  -moz-transform: translate(90%, -50%);
  -ms-transform: translate(90%, -50%);
  transform: translate(90%, -50%);
  right: 1px;
}

.emotion-3 span:nth-of-type(2) {
  left: 0;
  -webkit-transform: translate(-90%, -50%) rotate(45deg);
  -moz-transform: translate(-90%, -50%) rotate(45deg);
  -ms-transform: translate(-90%, -50%) rotate(45deg);
  transform: translate(-90%, -50%) rotate(45deg);
}

.emotion-3 span i {
  font-size: 6px!important;
}

.emotion-4 [class^='icon-'],
.emotion-4 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-5 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #27ae60;
}

.emotion-9 {
  height: 20px;
  font-size: 10px;
  border-radius: 4px;
  padding-inline: 6px;
}

.emotion-9.emotion-9 .ant-input-prefix {
  -webkit-margin-end: 6px;
  margin-inline-end: 6px;
}

.emotion-9.emotion-9 i {
  font-size: 12px!important;
}

.emotion-9[type='number']::-webkit-inner-spin-button,
.emotion-9 input[type='number']::-webkit-inner-spin-button {
  display: none;
}

.emotion-9.ant-input {
  font-weight: 500!important;
}

.emotion-9 .ant-input-prefix {
  -webkit-margin-end: 8px;
  margin-inline-end: 8px;
}

.emotion-9 i {
  color: #d9d9d9;
  -webkit-transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-9.ant-input-affix-wrapper-focused i {
  color: #27ae60;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) {
  background-color: #eb57570a;
  border-width: 1px;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) i {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i:before,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) i:before {
  content: '\\e91c';
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input {
  background: transparent;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-moz-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::-moz-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-status-error:not(.ant-input-disabled) input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input,
.emotion-9.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::placeholder,
.emotion-9.ant-input-status-error:not(.ant-input-disabled) input::placeholder {
  color: #eb5757;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
  width: 100%;
}

.emotion-10.emotion-10>button {
  width: 100%!important;
}

.emotion-10.emotion-10>button i {
  color: rgba(38, 35, 63, 0.88);
}

.emotion-12 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  background: #FFF;
  color: #26233F;
  border-color: #E9E8EE;
  padding: 3px 8px;
  -webkit-padding-start: 6px;
  padding-inline-start: 6px;
  border-radius: 4px;
  height: 20px;
  font-size: 10px;
  width: 20px!important;
}

.emotion-12>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-12.ant-btn.ant-btn-round {
  padding-inline: 8px;
}

.emotion-12 i {
  color: #27ae60;
}

.emotion-12.ant-btn-default:not(:disabled):not(.emotion-12.ant-btn-disabled):hover {
  background: #FFF;
  color: #47BA73;
  border-color: #E5E4EC;
}

.emotion-12.ant-btn-default:not(:disabled):not(.emotion-12.ant-btn-disabled):active {
  background: #FFF;
  color: #178749;
  border-color: #E1DFE8;
}

.emotion-12 i {
  font-size: 16px;
}

.emotion-12>span {
  gap: 4px;
}

.emotion-12,
.emotion-12 span {
  line-height: 10px;
}

.emotion-12.emotion-12>span>span>i {
  font-size: 12px;
}

.emotion-14 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
    data-testid="asdqwe"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-preview_filled emotion-5"
          />
        </span>
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-color-picker_filled emotion-5"
          />
        </span>
      </div>
    </div>
  </div>
  <input
    class="ant-input ant-input-sm emotion-8 ant-input-outlined emotion-9"
    max="360"
    min="0"
    placeholder="0"
    type="number"
    value="0"
  />
  <div
    class="emotion-10"
  >
    <button
      class="ant-btn emotion-8 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-12"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-arrow-rotate-left_outlined emotion-14"
          />
        </span>
      </span>
    </button>
    <button
      class="ant-btn emotion-8 ant-btn-default ant-btn-sm ant-btn-icon-only emotion-12"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          class="anticon undefined emotion-4"
        >
          <i
            class="icon-arrow-rotate-right_outlined emotion-14"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;
